plutus-tx/src/PlutusTx/List.hs

```diff
@@ -246,6 +248,27 @@ xs0 !! n0 = go n0 xs0
             then x
             else go (Builtins.subtractInteger n 1) xs
 
+-- | Index operator for builtin lists.
+--
+--   >>> indexBuiltinList (toBuiltin [10, 11, 12]) 2
+--   12
+--
+indexBuiltinList :: forall a. BI.BuiltinList a -> Integer -> a
+indexBuiltinList xs0 i0
+  | i0 `Builtins.lessThanInteger` 0 = traceError builtinListNegativeIndexError
+  | otherwise = go xs0 i0 ()
+  where
+    go :: BI.BuiltinList a -> Integer -> () -> a
```

Comment:
I wonder if making it an argument instead of just inlining it actually helps performance.

Reply:
I'm not sure I understand - making what an argument?

Reply:
`go :: BI.BuiltinList a -> Integer -> () -> a` vs `go :: BI.BuiltinList a -> Integer -> a`. Is that `()` needed? Does it help?

Reply:
Ah, I think you mean this:

```haskell
go :: BI.BuiltinList a -> Integer -> a
go xs i =
  Builtins.matchList
    xs
    (\_ -> traceError builtinListIndexTooLargeError)
    ( \hd tl _ ->
        if i `Builtins.equalsInteger` 0
          then hd
          else go tl (Builtins.subtractInteger i 1)
    )
    ()
```

I think they are the same but this does look prettier.